src/platform/vita/vita_input.c

```diff
@@ -0,0 +1,428 @@
+            if (right_x > slope * right_y)
+                right = 1;
+        }
+        // upper left quadrant
+        else if (right_y > 0 && right_x <= 0)
+        {
+            if (right_y > slope * (-right_x))
+                up = 1;
+            if ((-right_x) > slope * right_y)
+                left = 1;
+        }
+        // lower right quadrant
+        else if (right_y <= 0 && right_x > 0)
+        {
+            if ((-right_y) > slope * right_x)
+                down = 1;
+            if (right_x > slope * (-right_y))
+                right = 1;
+        }
+        // lower left quadrant
+        else if (right_y <= 0 && right_x <= 0)
+        {
+            if ((-right_y) > slope * (-right_x))
+                down = 1;
+            if ((-right_x) > slope * (-right_y))
+                left = 1;
+        }
+
+        if (!up_pressed && up) {
+            SDL_Event event;
+            event.type = SDL_KEYDOWN;
+            event.key.keysym.sym = SDLK_UP;
+            SDL_PushEvent(&event);
+        } else if (!down_pressed && down) {
+            SDL_Event event;
+            event.type = SDL_KEYDOWN;
+            event.key.keysym.sym = SDLK_DOWN;
+            SDL_PushEvent(&event);
+        }
+        if (!left_pressed && left) {
+            SDL_Event event;
+            event.type = SDL_KEYDOWN;
+            event.key.keysym.sym = SDLK_LEFT;
+            SDL_PushEvent(&event);
+        } else if (!right_pressed && right) {
+            SDL_Event event;
+            event.type = SDL_KEYDOWN;
+            event.key.keysym.sym = SDLK_RIGHT;
+            SDL_PushEvent(&event);
+        }
+    }
+}
+
+void vita_handle_virtual_keyboard()
+{
+    if (vkbd_requested) {
+        vkbd_requested = 0;
+        vita_start_text_input("", 0);
+    }
+}
+
+static void vita_start_text_input(char *initial_text, int multiline)
+{
+    char *text = vita_keyboard_get("Enter New Text:", initial_text, 600, multiline);
+    if (text != NULL)  {
+        for (int i = 0; i < 600; i++) {
+            SDL_Event down_event;
+            down_event.type = SDL_KEYDOWN;
+            down_event.key.keysym.sym = SDLK_BACKSPACE;
+            down_event.key.keysym.mod = 0;
+            SDL_PushEvent(&down_event);
+            SDL_Event up_event;
+            up_event.type = SDL_KEYUP;
+            up_event.key.keysym.sym = SDLK_BACKSPACE;
+            up_event.key.keysym.mod = 0;
+            SDL_PushEvent(&up_event);
+        }
+        for (int i = 0; i < 600; i++) {
+            SDL_Event down_event;
+            down_event.type = SDL_KEYDOWN;
+            down_event.key.keysym.sym = SDLK_DELETE;
+            down_event.key.keysym.mod = 0;
+            SDL_PushEvent(&down_event);
+            SDL_Event up_event;
+            up_event.type = SDL_KEYUP;
+            up_event.key.keysym.sym = SDLK_DELETE;
+            up_event.key.keysym.mod = 0;
+            SDL_PushEvent(&up_event);
+        }
+        int i=0;
+        while (text[i]!=0 && i<599) {
```

Comment:
```suggestion
        for (int i=0; i < 599 && text[i] != 0; i++) {
```

Or maybe
```suggestion
        for (int i=0; i < 599; i++) {
            if (text[i] == 0)
                break;
```

Reply:
Fixed in 35f42ab00e2b28a49e966a752eb3016a334d53bb